Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/PokePointer.cs

```diff
@@ -93,6 +115,29 @@ public override void OnPreSceneQuery()
             visuals.SetActive(IsActive);
         }
 
+        private bool FindClosestTouchableForLayerMask(LayerMask layerMask, out ColliderNearInteractionTouchable closest, out float distance, out Vector3 normal)
+        {
+            closest = null;
+            distance = float.PositiveInfinity;
+            normal = Vector3.zero;
+
+            Collider[] colliders = UnityEngine.Physics.OverlapSphere(Position, touchableDistance, layerMask, triggerInteraction);
```

Comment:
You can do this without allocations: https://docs.unity3d.com/ScriptReference/Physics.OverlapSphereNonAlloc.html

Have we profiled this stuff for speed etc?

Reply:
I have read conflicting explanations of the OverlapSphereNonAlloc variant regarding resizing of the colliders buffer. There is not a great way to know the number of potential touchables we could find, so we can't allocate the buffer reliably beforehand. Some commenters said that it gets resized if the buffer is too small, but most say that you will miss potential colliders if the buffer is too small. We could of course just make this a configurable profile setting with a reasonable default (say 64) and print a warning if the buffer ever gets full.

Reply:
>Some commenters said that it gets resized if the buffer is too small, but most say that you will miss potential colliders if the buffer is too small. 

I can test this to confirm but my understanding was that you'll miss colliders. The amount of garbage we'll generate with the non-alloc version is brutal so the overhead of a huge buffer is worth it. Making it a profile setting sounds ideal.

Reply:
Fixed, i'm using the NonAlloc variant now.